### PR TITLE
Stories/356 email setup docs

### PR DESCRIPTION
**Pull request by jessesnyder**
## Description
- Revises and expands documentation for setting up email notifications in Dallinger.
- Moves documentation to the User/Experimenter section of the docs.

## Motivation and Context
Email setup is complicated and full of pitfalls. It benefits from thorough documentation.

## How Has This Been Tested?
Documentation was based on several attempts at setting up and sending email notifications using Gmail and GMX Mail services.

### CODE REVIEW

File: docs/source/email_setup.rst [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+
+The Config Settings
+~~~~~~~~~~~~~~~~~~~
+
+``smtp_host`` The hostname and port of the SMTP (outgoing email) server through
+which all email will be sent. This defaults to `smtp.gmail.com:587`, the Google
+SMTP server. If you want to send email *from* a Gmail address, or a custom
+domain set up to use Gmail for email, this default setting is what you want.
+
+``smtp_username`` The username with which to log into the SMPT server, which
```
Comment (witekdev) [inline docs/source/email_setup.rst]: SMPT -> SMTP

File: docs/source/email_setup.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+cause things to behave differently than expected:
+
+
+.. _two-factor-auth:
+
+Two-Factor Authentication
+"""""""""""""""""""""""""
+
+Having two-factor authentication enabled for the outgoing email account will
+prevent dallinger from sending email without some additional steps. Details
```
Comment (witekdev) [inline docs/source/email_setup.rst]: Detailed instructions vs Details instructions?

File: docs/source/email_setup.rst [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+instructions are provided for Gmail, below. Other email services which support
+two-factor authentication may provide equivalent solutions for marking an app
+as safe to send email.
+
```
Comment (witekdev) [inline docs/source/email_setup.rst]: The grammar is a bit funny (to me) here?
Other email services which support two-factor authentication may provide equivalent solutions for marking an app
as safe to send email.

File: docs/source/email_setup.rst [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+#. Find and immediately copy the password under *Your app passwords*. Type or paste the
+   password into the `.dallingerconfig` file in your home directory.
+   You will not be able to view the password again, so if you miss it, you'll
+   need to delete the one you just created and create a new one.
+#. Click *Done*.
+
+Firewall/antivirus
+""""""""""""""""""
+When developing locally, antivirus or firewall software may prevent outgoing
+email from being sent, and cause dallinger to raise a `socket.timeout` error.
```
Comment (witekdev) [inline docs/source/email_setup.rst]: Maybe keep Dallinger as the name of the app consistently as "Dallinger" vs "dallinger" throughout the document?
Occurs in several places.

File: docs/source/email_setup.rst [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+Temporarily disabling these tools is the easiest workaround.
+
+Google "Less secure apps"
+"""""""""""""""""""""""""
+If you do **not** have two-factor authentication enabled, Gmail may require that
+you enable "less secure apps" in order to send email from dallinger. You will
+likely know you are encountering this problem because you will receieve warning
```
Comment (witekdev) [inline docs/source/email_setup.rst]: Typo - "receieve"

File: docs/source/email_setup.rst [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+-------------
+Sending email from Dallinger requires 5 configuration settings, described in
+turn below. Like all configuration settings, they can be set up in either
+`.dallingerconfig` in your home directory, or in `config.txt` in the root
+directory of your experiment.
+
+The Config Settings
+~~~~~~~~~~~~~~~~~~~
+
+``smtp_host`` The hostname and port of the SMTP (outgoing email) server through
```
Comment (witekdev) [inline docs/source/email_setup.rst]: I would put the 'smtp_host', 'smtp_username' on its own line, like is done in the configuration text.

File: docs/source/email_setup.rst [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+which all email will be sent. This defaults to `smtp.gmail.com:587`, the Google
+SMTP server. If you want to send email *from* a Gmail address, or a custom
+domain set up to use Gmail for email, this default setting is what you want.
+
+``smtp_username`` The username with which to log into the SMPT server, which
+will very likely be an email address (if you are using a Gmail address to send
+email, you will use that address for this value).
+
+``smpt_password`` The password associated with the ``smtp_username``. **NOTE**
```
Comment (witekdev) [inline docs/source/email_setup.rst]: Suggestion: separate the NOTE section with a line-break for more visibility?

File: docs/source/email_setup.rst (hunk 3 above)
Comment (jessesnyder) [reply docs/source/email_setup.rst]: I agree. It became tortured in part because I was trying to avoid the problematic phrase "whitelist". Will rework.

Comment (jessesnyder) [reply docs/source/email_setup.rst]: Just going to end the sentence at "... solutions."

**Pull request merged**